meteor/client/ui/Settings/ShowStyle/VariantListItem.tsx

```diff
@@ -0,0 +1,175 @@
+import { EditAttribute } from '../../../lib/EditAttribute'
+import { ShowStyleVariant, ShowStyleVariants } from '../../../../lib/collections/ShowStyleVariants'
+import { ConfigManifestSettings } from '../ConfigManifestSettings'
+import { ConfigManifestEntry, SourceLayerType } from '@sofie-automation/blueprints-integration'
+import { MappingsExt } from '@sofie-automation/corelib/dist/dataModel/Studio'
+import { ProtectedString, unprotectString } from '@sofie-automation/shared-lib/dist/lib/protectedString'
+import { ShowStyleBase } from '../../../../lib/collections/ShowStyleBases'
+import { MeteorCall } from '../../../../lib/api/methods'
+import { logger } from '../../../../lib/logging'
+import { doModalDialog } from '../../../lib/ModalDialog'
+import { TFunction } from 'react-i18next'
+import { i18n } from 'i18next'
+
+interface IShowStyleVariantItemProps {
+	index: number
+	showStyleVariant: ShowStyleVariant
+	showStyleBase: ShowStyleBase
+	dndVariants: ShowStyleVariant[]
+	blueprintConfigManifest: ConfigManifestEntry[]
+	layerMappings?: { [key: string]: MappingsExt }
+	sourceLayers?: Array<{ name: string; value: string; type: SourceLayerType }>
+
+	t: TFunction<'translation', undefined>
+	i18n: i18n
+	tReady: boolean
+}
+
+export const VariantListItem: React.FunctionComponent<IShowStyleVariantItemProps> = (
+	props: IShowStyleVariantItemProps
+) => {
+	const { t } = props
+	const initialEditedMappings: ProtectedString<any>[] = []
+	const [editedMappings, setEditedMappings] = useState(initialEditedMappings)
+
+	function copyShowStyleVariant(showStyleVariant: ShowStyleVariant): void {
+		showStyleVariant.name = `Copy of ${showStyleVariant.name}`
+		showStyleVariant._rank = props.dndVariants.length
+		MeteorCall.showstyles.importShowStyleVariantAsNew(showStyleVariant).catch(logger.warn)
+	}
+
+	function downloadShowStyleVariant(showStyleVariant: ShowStyleVariant): void {
+		const showStyleVariants = [showStyleVariant]
+		const jsonStr = JSON.stringify(showStyleVariants)
+		const fileName = `${showStyleVariant.name}_showstyleVariant_${showStyleVariant._id}.json`
+		download(jsonStr, fileName)
+	}
+
+	function download(jsonStr: string, fileName: string): void {
+		const element = document.createElement('a')
+		element.href = URL.createObjectURL(new Blob([jsonStr], { type: 'application/json' }))
+		element.download = fileName
+
+		element.click()
+	}
+
+	function isItemEdited(layerId: ProtectedString<any>): boolean {
+		return editedMappings.indexOf(layerId) >= 0
+	}
+
+	function finishEditItem(layerId: ProtectedString<any>): void {
+		const index = editedMappings.indexOf(layerId)
+		if (index >= 0) {
+			editedMappings.splice(index, 1)
+			setEditedMappings([...editedMappings])
+		}
+	}
+
+	function editItem(layerId: ProtectedString<any>): void {
+		if (editedMappings.indexOf(layerId) < 0) {
+			editedMappings.push(layerId)
+			setEditedMappings([...editedMappings])
+		} else {
+			finishEditItem(layerId)
+		}
```

Comment:
You could use `isItemEdited` and "move" the push into the spreading of `editedMappings`:
```suggestion
		if (this.isItemEdited(layerId)) {
			finishEditItem(layerId)
			return
		}
		setEditedMappings([...editedMappings, layerId])
```